### ISSUE TITLE
router does not permit charset declaration on content-type

### ISSUE THREAD

**Issue by mdaniel** [toshi-search/Toshi#111]
## What happened

A mime type containing the `;charset=` qualifier caused Toshi to respond with `400 Bad Request` to an index creation request

## What was expected

The index would have been created. Many, _many_ http libraries will send along the charset if they know it, because it's polite

## How to reproduce

Assuming you have run `cargo test`, there will be `new_index` containing the tantivy index, and its schema, in the current directory; thus:

```console
$ jq .schema new_index/meta.json | \
    curl --compress -vH 'content-type: application/json;charset=utf-8' -X PUT -d @-  127.0.0.1:9200/foo/_create
```
produces:
```
> PUT /foo/_create HTTP/1.1
> Host: 127.0.0.1:9200
> User-Agent: curl/7.54.0
> Accept: */*
> Accept-Encoding: deflate, gzip
> content-type: application/json;charset=utf-8
>
< HTTP/1.1 400 Bad Request
< content-type: application/json
< content-encoding: deflate
< transfer-encoding: chunked
< date: Sun, 20 Jan 2019 20:58:04 GMT
<
{"message":"Bad request","uri":"/foo/_create"}
```
but the unqualified `content-type` header creates the index as expected
```console
$ jq .schema new_index/meta.json | \
    curl --compress -vH 'content-type: application/json' -X PUT -d @-  127.0.0.1:9200/foo/_create
```

**Comment by hntd187** [comment]
Hey, thanks for this. I'm actually not entirely sure if this is a tower-web issue or if it's something related to my impl. Lemme poke around with this and ask Carl if maybe he has an idea on if it's something we can fix or if it's a tower issue.

**Comment by hntd187** [comment]
This should be fixed from #128 thanks 😄
